src/compiler/transformers/module/module.ts

```diff
@@ -378,6 +383,10 @@ namespace ts {
             // Append the 'export =' statement if provided.
             addExportEqualsIfNeeded(statements, /*emitAsReturn*/ true);
 
+            if (shouldAppendUnderscoreUnderscoreEsModule) {

```

Comment:
We didn't set `shouldAppendUnderscoreUnderscoreEsModule` here. This is after we have passed through transform module already and we just check if anyone set the flag